src/Service.Tests/Unittests/SqlQueryExecutorUnitTests.cs

```diff
@@ -318,6 +318,62 @@ Mock<MsSqlQueryExecutor> queryExecutor
+        /// was invoked the expected number of times.
+        /// </summary>
+        [DataTestMethod, TestCategory(TestCategory.MSSQL)]
+        [DataRow(false, 4, DisplayName = "Data should be successfully read when amount of data being read is less than max-db-response-size-mb")]
+        [DataRow(false, 5, DisplayName = "Data should be successfully read when amount of data being read is less than max-db-response-size-mb")]
+        [DataRow(true, 6, DisplayName = "Exception should be thrown when amount of data read is more than max-db-response-size-mb")]
+        public void ValidateStreamingLogicAsync(bool exceptionExpected, int readDataLoops)
+        {
+            TestHelper.SetupDatabaseEnvironment(TestCategory.MSSQL);
+            FileSystem fileSystem = new();
+            FileSystemRuntimeConfigLoader loader = new(fileSystem);
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: new DataSource(DatabaseType: DatabaseType.MSSQL, "", Options: null),
+                Runtime: new(
+                        Rest: new(),
+                        GraphQL: new(),
+                        Host: new(Cors: null, Authentication: null, MaxResponseSizeMB: 5)
+                    ),
+                Entities: new(new Dictionary<string, Entity>()));
+
+            RuntimeConfigProvider runtimeConfigProvider = TestHelper.GenerateInMemoryRuntimeConfigProvider(runtimeConfig);
+
+            Mock<ILogger<QueryExecutor<SqlConnection>>> queryExecutorLogger = new();
+            Mock<IHttpContextAccessor> httpContextAccessor = new();
+            DbExceptionParser dbExceptionParser = new MsSqlDbExceptionParser(runtimeConfigProvider);
+
+            // Instantiate the MsSqlQueryExecutor and Setup parameters for the query
+            MsSqlQueryExecutor msSqlQueryExecutor = new(runtimeConfigProvider, dbExceptionParser, queryExecutorLogger.Object, httpContextAccessor.Object);
+
+            try
+            {
+                /// In this test the DbDataReader.GetChars method is mocked to return 1024*1024 bytes (1MB) of data.
+                /// Max available size is set to 5 MB.
+                /// Based on number of loops, the data read will be 1MB * readDataLoops.Exception should be thrown in test cases where we go above 5MB.
+                Mock<DbDataReader> dbDataReader = new();
+                dbDataReader.Setup(d => d.HasRows).Returns(true);
+                dbDataReader.Setup(x => x.GetChars(It.IsAny<int>(), It.IsAny<long>(), It.IsAny<char[]>(), It.IsAny<int>(), It.IsAny<int>())).Returns(1024 * 1024);
+                int availableSize = (int)runtimeConfig.MaxResponseSizeMB() * 1024 * 1024;
+                for (int i = 0; i < readDataLoops; i++)
+                {
+                    availableSize -= msSqlQueryExecutor.StreamData(dbDataReader.Object, availableSize, new());
```

Comment:
named params

Reply:
when real QueryExecutor.StreamData(..) is called, would you be passing a new StringBuilder multiple times in a single request? or reusing the same stringbuilder?

Reply:
fixed. We would be reusing the same stringbuilder.